--- a/bzl/valdi/valdi_android_application.bzl
+++ b/bzl/valdi/valdi_android_application.bzl
@@ -99,6 +99,10 @@
         assets = assets,
         assets_dir = assets_dir,
         resource_files = resource_files,
+        proguard_specs = [
+            "@valdi//src:client_proguard-rules",
+            "//:my_project_proguard.pro",
+        ],
         tags = ["valdi_android_application"],
         deps = [
             ":{}".format(src_target),
